Bokeh doesn't work with Tornado master

```pytz
distributed/bokeh/core.py:7: in <module>
    from bokeh.server.server import Server
../../Software/anaconda/lib/python3.6/site-packages/bokeh/server/server.py:35: in <module>
    from .tornado import BokehTornado, DEFAULT_WEBSOCKET_MAX_MESSAGE_SIZE_BYTES
../../Software/anaconda/lib/python3.6/site-packages/bokeh/server/tornado.py:24: in <module>
    from .urls import per_app_patterns, toplevel_patterns
../../Software/anaconda/lib/python3.6/site-packages/bokeh/server/urls.py:39: in <module>
    from .views.ws import WSHandler
../../Software/anaconda/lib/python3.6/site-packages/bokeh/server/views/ws.py:20: in <module>
    from ...protocol.receiver import Receiver
../../Software/anaconda/lib/python3.6/site-packages/bokeh/protocol/receiver.py:8: in <module>
    from tornado.concurrent import return_future
E   ImportError: cannot import name 'return_future'
```

Also cc @bdarnell in case he's interested in downstream testing

--- a/bokeh/protocol/receiver.py
+++ b/bokeh/protocol/receiver.py
@@ -5,7 +5,7 @@ message objects that can be processed.
 from __future__ import absolute_import
 
 import six
-from tornado.concurrent import return_future
+from tornado import gen
 
 from .exceptions import ValidationError
 
@@ -70,8 +70,8 @@ class Receiver(object):
         self._message = None
         self._buf_header = None
 
-    @return_future
-    def consume(self, fragment, callback=None):
+    @gen.coroutine
+    def consume(self, fragment):
         ''' Consume individual protocol message fragments.
 
         Args:
@@ -80,12 +80,9 @@ class Receiver(object):
                 assembled, the receiver state will reset to begin consuming a
                 new message.
 
-            callback (callable, optional)
-                Argument required by ``return_future`` decorator
-
         '''
         self._current_consumer(fragment)
-        callback(self._message)
+        raise gen.Return(self._message)
 
     def _HEADER(self, fragment):
         self._assume_text(fragment)

